modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/KeyTypeFilter/index.js

```diff
@@ -0,0 +1,189 @@
+/* eslint-disable no-unused-vars */
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import ClayAlert from '@clayui/alert';
+import ClayButton from '@clayui/button';
+import {ClayCheckbox, ClayInput} from '@clayui/form';
+import classNames from 'classnames';
+import {useEffect, useState} from 'react';
+
+const INVALID_NODE_MESSAGE = 'Enter a valid number';
+const INVALID_MIN_NODE_MESSAGE = 'Enter a minimum node value greater than 0';
+const INVALID_MAX_NODE_MESSAGE = 'Max nodes must be greater than min nodes';
+const INVALID_NEGATIVE_NODES_MESSAGE = 'Enter nodes values greater than 0';
+
+const KeyTypeFilter = ({clearInputs, hasVirtualCluster, setFilters}) => {
+	const [minNodesValue, setMinNodesValue] = useState('');
+	const [maxNodesValue, setMaxNodesValue] = useState('');
+
+	const [virtualClusterChecked, setVirtualClusterChecked] = useState(false);
+	const [onPromiseChecked, setOnPromiseChecked] = useState(false);
+
+	const [errorMessage, setErrorMessage] = useState('');
+
+	useEffect(() => {
+		if (isNaN(minNodesValue) || isNaN(maxNodesValue)) {
+			setErrorMessage(INVALID_NODE_MESSAGE);
+		}
+		else if (minNodesValue === '0') {
+			setErrorMessage(INVALID_MIN_NODE_MESSAGE);
+		}
+		else if (maxNodesValue <= -1 || minNodesValue <= -1) {
+			setErrorMessage(INVALID_NEGATIVE_NODES_MESSAGE);
+		}
+		else if (maxNodesValue < minNodesValue) {
+			setErrorMessage(INVALID_MAX_NODE_MESSAGE);
+		}
+		else {
+			setErrorMessage('');
+		}
```

Comment:
Instead of using else ifs, just do a `return` to avoid that
```suggestion
		if (isNaN(minNodesValue) || isNaN(maxNodesValue)) {
			setErrorMessage(INVALID_NODE_MESSAGE);
			return;
		}
		if (minNodesValue === '0') {
			setErrorMessage(INVALID_MIN_NODE_MESSAGE);
			return;
		}
		 if (maxNodesValue <= -1 || minNodesValue <= -1) {
			setErrorMessage(INVALID_NEGATIVE_NODES_MESSAGE);
			return;
		}
		if (maxNodesValue < minNodesValue) {
			setErrorMessage(INVALID_MAX_NODE_MESSAGE);
			return;
		}
	
			setErrorMessage('');
		
```

Reply:
done!